pkg/kubectl/resource_printer.go

```diff
@@ -2196,6 +2196,10 @@ func printConfigMapList(list *api.ConfigMapList, w io.Writer, options PrintOptio
 func printPodSecurityPolicy(item *extensions.PodSecurityPolicy, w io.Writer, options PrintOptions) error {
 	name := formatResourceName(options.Kind, item.Name, options.WithKind)
 
+	if options.WithNamespace {
```

Comment:
Yeah, this is becoming a (not nice) trend. Has issues with lists, missing new line separators in some scenarios, boilerplate code, etc. 

I'd say this should happen somewhere in `PrintObj`, however we need to check how doable that is - we don't seem to have any type or object meta that clearly states a given type is not namespaced (other than checking an empty `namespace` field in the actual object meta, and the `nonNamespaced` annotation used by `client-gen`).